src/systems/fem_system.C

```diff
@@ -288,16 +294,27 @@ void add_element_system(const FEMSystem & _sys,
           (_femcontext.get_elem_jacobian(),
            _femcontext.get_elem_residual(),
            _femcontext.get_dof_indices(), false);
+      else if (_no_constraints)
+	{
+	  // Do noting
+	}
       else
         _sys.get_dof_map().constrain_element_vector
           (_femcontext.get_elem_residual(), _femcontext.get_dof_indices(), false);
     }
   else if (_get_jacobian)
     {
-      // Heterogeneous and homogeneous constraints are the same on the
-      // matrix
-      _sys.get_dof_map().constrain_element_matrix
-        (_femcontext.get_elem_jacobian(), _femcontext.get_dof_indices(), false);
+      if (_no_constraints)
+	{
+	  // Do noting
```

Comment:
Not sure I understand, should I just get rid of that entire if(_no_constraints) block ?